src/Board.js

```diff
@@ -1,5 +1,172 @@
 /* Represents the board state at a point in time */
 class Board 
 {
+	constructor(adjacencyMatrix)
+	{
+		this.cells = adjacencyMatrix;
+		this.contents = [];
+		for (let i = 0; i < this.cells.length; i++)
+		{
+			this.contents.push(undefined);
+		}
+	}
 	
+	/**
+	 *  Returns a Set of all locations (as indices) described
+	 *  by the vector relative to startLocation. Does not account
+	 *  for a blocked destination, since this could be used for
+	 *  capture, but it will not continue down a blocked path.
+	 *  
+	 *  Always returns a Set. Returns an empty Set if no such
+	 *  locations can be found. 
+	 */
+	GetCellIndices(vector, startLocation)
+	{
+		const allCellIndices = new Set();
+		
+		const x = vector.components[0];
+		const y = vector.components[1];
+		
+		let dx = 0;
+		let dy = 0;
+		for (let i = 0; i < x.maxRep; i++)
+		{
+			dx += x.length;
+			for (let j = 0; j < y.maxRep; j++)
+			{
+				dy += y.length;
+				const output = this.GetPathOutput(startLocation, dx, dy, x.hop, y.hop, x.jump, y.jump);
+				
+				// TODO: Update second clause so this function works for capture and for movement
+				if (output == -1 || this.contents[output] != undefined)
+					continue;
+				allCellIndices.add(output);
+			}
+		}
+		
+		return allCellIndices;
+	}
+	
+	/**
+	 *  Returns the index of the cell dx and dy units
+	 *  away from start. Returns -1 if no such 
+	 *  destination exists. 
+	 */
+	GetPathOutput(start, dx, dy, xHop, yHop, xJump, yJump)
+	{
+		let destination = start;
+		let previous = start;
+		let stepX = 0;
+		let stepY = 0;
+			
+		while (dx != 0 || dy != 0)
+		{
+			stepX = 0;
+			stepY = 0;
+			if (dx < 0)
+			{
+				stepX = -1;
+				dx++;
+			}
+			if (dx > 0)
+			{
+				var stepX = 1;
+				dx--;
+			}
+			if (dy < 0)
+			{
+				var stepY = 1;
+				dy++;
+			}
+			if (dy > 0)
+			{
+				var stepY = -1;
+				dy--;
+			}
+			
+			/* Add 1 because arrays cannot have negative indices */
+			const direction = (stepY+1) * 3 + (stepX+1);
```

Comment:
It's based on our dimensions. I'll update so it's more clear.

Reply:
The plus 1 is so that it comes out, at minimum, equal to zero, which is the smallest index an array will support. That comment could probably be improved.